backend/src/main/kotlin/org/loculus/backend/utils/MetadataEntry.kt

```diff
@@ -60,40 +67,44 @@ fun revisionEntryStreamAsSequence(metadataInputStream: InputStream): Sequence<Re
         CSVFormat.TDF.builder().setHeader().setSkipHeaderRecord(true).build(),
     )
 
-    if (!csvParser.headerNames.contains(HEADER_TO_CONNECT_METADATA_AND_SEQUENCES)) {
-        throw UnprocessableEntityException(
-            "The revised metadata file does not contain the header '$HEADER_TO_CONNECT_METADATA_AND_SEQUENCES'",
+    val headerNames = csvParser.headerNames
+    val submissionIdHeaders = listOf(
+        HEADER_TO_CONNECT_METADATA_AND_SEQUENCES,
+        HEADER_TO_CONNECT_METADATA_AND_SEQUENCES_ALTERNATE_FOR_BACKCOMPAT,
+    ).filter { headerNames.contains(it) }
```

Comment:
This is really hard to read

Reply:
We have separate for RevisionEntrys and MetadataEntrys - I'm happy to accept any pushes to refactor this but I only think we need to keep the back-compatibility for a few weeks anyway

Reply:
I extracted a shared function